build-system/eslint-rules/no-module-exports.js

```diff
@@ -0,0 +1,46 @@
+/**
+ * Copyright 2018 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+module.exports = {
+  meta: {
+    type: "suggestion",
+    schema: [],
+    messages: {
+      unexpected: "Unexpected use of `module.exports` statement."
+    }
+  },
+
+  create(context) {
+
+    function isModuleExports(node) {
+      return node.object.type === 'Identifier' && (/^(module.exports)$/).test(node.object.name)
```

Comment:
This is incorrect. `module.exports` is a `MemberExpression` composed of two `Identifiers`, not a plain `Identifier`. Please see https://astexplorer.net/#/gist/f81b69fca8b30a5592b4ef13cde5fff8/096f72d923b5aaa0e7d1470037bdc2403af177f5, which is essentially a ESLint repl to help build plugins.

Reply:
Got it, thanks for the link. Submitted fix #19581